packages/react-dom/src/__tests__/ReactTestUtilsAct-test.js

```diff
@@ -68,6 +82,26 @@ function runActTests(label, render, unmount) {
       document.body.removeChild(container);
     });
     describe('sync', () => {
+      it('warns if an effect is queued outside an act scope, except in legacy sync+non-strict mode', () => {
+        function App() {
+          React.useEffect(() => {}, []);
+          return null;
+        }
+        expect(() => {
+          render(<App />, container);
+          // flush all queued work
+          Scheduler.unstable_flushAll();
+        }).toWarnDev(
+          label !== 'legacy sync mode'
```

Comment:
:headdesk: This is why I don't like abstractions in tests

How about we write a separate test for each mode? :D